### PR TITLE
Fixes various typos with codespell

### PR DESCRIPTION
**Pull request by lerni**
Mostly fixes typos in strings, docs & comment

Also some variablenames are renamed:
- `famework/src/Core/CoreKernel.php` `$environment`
- `vendor/silverstripe/framework/src/ORM/DataObject.php` `$identifier`
- `vendor/silverstripe/framework/tests/php/Forms/GridField/GridFieldSortableHeaderTest.php` `$component`

Had a play with [codespell](https://github.com/codespell-project/codespell) - this PR is the outcome. Hopefully someone is willing to review - it's a lot.

### CODE REVIEW

**Comment by michalkleiner** [comment]
Thanks @lerni, nice job. Been thinking of going through the codebase of the core modules at some point, too.

I've started reviewing the changes, got to 30 files so far, the majority is looking good. Should get through the rest today.

File: src/Core/Injector/Injector.php [hunk 1]
```diff
@@ -201,7 +201,7 @@ class Injector implements ContainerInterface
     const SINGLETON = 'singleton';
 
     /**
-     * Specif ya service type prototype
+     * Specify ya service type prototype
```
Comment (michalkleiner) [inline src/Core/Injector/Injector.php:204]:
```suggestion
     * Specify a service type prototype
```

File: tests/php/Forms/GridField/GridFieldSortableHeaderTest.php [hunk 2]
```diff
@@ -43,7 +43,7 @@ public function testRenderHeaders()
         $form = new Form(null, 'Form', new FieldList(), new FieldList());
         $gridField = new GridField('testfield', 'testfield', $list, $config);
         $gridField->setForm($form);
-        $compontent = $gridField->getConfig()->getComponentByType(GridFieldSortableHeader::class);
+        $component = $gridField->getConfig()->getComponentByType(GridFieldSortableHeader::class);
         $htmlFragment = $compontent->getHTMLFragments($gridField);
```
Comment (michalkleiner) [inline tests/php/Forms/GridField/GridFieldSortableHeaderTest.php:47]:
```suggestion
        $htmlFragment = $component->getHTMLFragments($gridField);
```

File: tests/php/Forms/GridField/GridFieldSortableHeaderTest.php [hunk 3]
```diff
@@ -77,7 +77,7 @@ public function testGetManipulatedData()
         $state->SortColumn = 'City';
         $state->SortDirection = 'asc';
 
-        $compontent = $gridField->getConfig()->getComponentByType(GridFieldSortableHeader::class);
+        $component = $gridField->getConfig()->getComponentByType(GridFieldSortableHeader::class);
         $listA = $compontent->getManipulatedData($gridField, $list);
```
Comment (michalkleiner) [inline tests/php/Forms/GridField/GridFieldSortableHeaderTest.php:81]:
```suggestion
        $listA = $component->getManipulatedData($gridField, $list);
```

**Review by michalkleiner — CHANGES_REQUESTED** [review]
Ok, went over all the changes.

@lerni the change in `GridFieldSortableHeaderTest` needs further work, the PR only renames the initial variable when created, but not when used. I've added two change suggestions, but you'd need to go through the class and update all the occurrences.

**Comment by michalkleiner** [comment]
Three minor things to confirm with @maxime-rainville or @emteknetnz:
1. The change to `CoreKernel` about renaming the protected `$enviroment` variable. Within the framework, I'd say this change is fine, but people might have their own kernels extending CoreKernel and not using the getter/setter methods. Though targetting main I'd be happy to mention this in the docs and keep the fix.
2. Fixing typos in changelogs where they were basically generated from commit messages including the same typos, so it would become disconnected, though the commit hash still stays valid. For better legibility of past changelogs, I'd be for keeping the spelling fixes, I don't expect people searching the history of the repo for exact commit message matches.
3. Inner-method variable name change in `DataObject` is fine with me, but a confirmation would be good.

**Comment by emteknetnz** [comment]
Yup agree with what @michalkleiner has said re 1. and 2.  3. is fine with the rename

**Review by michalkleiner — APPROVED** [review]
Thanks @lerni, appreciate the contribution.

**Comment by lerni** [comment]
@michalkleiner thank you for reviewing it so fast: I'm impressed! :100: :heart: May this motivates me checking others like CMS with codespell as well :wink:

**Pull request merged**